### PR TITLE
tsp, in test, use union as extensible enum

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
First update test to use union as "extensible enum".

For now, enum is still as "extensible enum". We cannot break those on SDK repo.
We can change that behavior (switch to treat enum as "fixed enum") only after all lib and spec is updated, and the commit ID on SDK repo is also updated.

### CODE REVIEW

File: typespec-tests/tsp/enum.tsp [hunk 1]
```diff
@@ -19,10 +19,11 @@ enum Color {
...
 }
 
-enum ColorModel {
-  Red,
-  Blue,
-  Green,
+union ColorModel {
+  string,
+  Red: "Red",
+  Blue: "Blue",
```
Comment (haolingdong-msft) [inline typespec-tests/tsp/enum.tsp:25]: why is this key value pairs? i.e. `Red: "Red"`, but others are not?

Comment (weidongxu-microsoft) [reply typespec-tests/tsp/enum.tsp:25]: There is a line ref to that UnionVariant
```
  colorModelValue: ColorModel.Blue;
```

so it needs a name.

File: typespec-tests/tsp/enum.tsp [hunk 2]
```diff
@@ -71,9 +72,10 @@ interface EnumOp {
   @route("/color")
   getColor(): Color;
 
-  @get
-  @route("/colormodel")
-  getColorModel(): ColorModel;
+  // TODO: the response as Union is ambiguous
+  // @get
+  // @route("/colormodel")
+  // getColorModel(): ColorModel;
```
Comment (weidongxu-microsoft) [inline typespec-tests/tsp/enum.tsp:78]: Disabled for now.

This op now equivalent to
```
getColorModel(): string | "Red" | "Blue" | "Green";
```
this union in response looks a bit strange (e.g. if compare to `getColorModel(): OkResponse | NoContentResponse`)

File: typespec-tests/tsp/enum.tsp (hunk 1 above)
Comment (haolingdong-msft) [reply typespec-tests/tsp/enum.tsp:25]: I see.

File: typespec-tests/tsp/enum.tsp (hunk 2 above)
Comment (weidongxu-microsoft) [reply typespec-tests/tsp/enum.tsp:78]: I guess I still need to solve this now... cadl-ranch has such test case.

Comment (weidongxu-microsoft) [reply typespec-tests/tsp/enum.tsp]: Changed to explicit `@body` on Timothee's advice.

File: typespec-tests/tsp/lro.tsp [hunk 3]
```diff
@@ -46,11 +46,12 @@ namespace Cadl.LongRunning {
...
   @lroStatus
-  enum JobStatus {
+  union JobStatus {
+    string,
     NotStarted: "notStarted",
     Running: "running",
-    Succeeded: "succeeded",
-    Failed: "failed",
+    Succeeded: "Succeeded",
+    Failed: "Failed",
```
[the remark this answers is not in the document]
Comment (haolingdong-msft) [reply typespec-tests/tsp/lro.tsp:54]: yeah, a bit strange to be case sensitive.

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**